client/js/components/GameView/GameDialogs.vue

```diff
@@ -0,0 +1,189 @@
+import FourDialog from '@/components/GameView/FourDialog.vue';
+import GameOverDialog from '@/components/GameView/GameOverDialog.vue';
+import ReauthenticateDialog from '@/components/GameView/ReauthenticateDialog.vue';
+import SevenDoubleJacksDialog from '@/components/GameView/SevenDoubleJacksDialog.vue';
+import ThreeDialog from '@/components/GameView/ThreeDialog.vue';
+
+export default {
+  name: 'GameDialogs',
+  components: {
+    CannotCounterDialog,
+    CounterDialog,
+    FourDialog,
+    GameOverDialog,
+    ReauthenticateDialog,
+    SevenDoubleJacksDialog,
+    ThreeDialog,
+  },
+  props: {
+    clearSelection: {
+      type: Function,
+      required: true,
+    },
+    game: {
+      type: Object,
+      required: true,
+    },
+    handleError: {
+      type: Function,
+      required: true,
+    },
+    resolvingSeven: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  computed: {
+    ...mapGetters(['player', 'opponent', 'opponentPointTotal', 'opponentQueenCount']),
+    discarding() {
+      return this.$store.state.game.discarding;
+    },
+    gameIsOver() {
+      return this.$store.state.game.gameIsOver;
+    },
+    hasTwoInHand() {
+      return this.twosInHand.length > 0;
+    },
+    myTurnToCounter() {
+      return this.game.myTurnToCounter;
+    },
+    // TODO: Refactor and combine usage in GameView.vue
+    mustReauthenticate: {
+      get() {
+        return this.$store.state.auth.mustReauthenticate;
+      },
+      set(val) {
+        this.$store.commit('setMustReauthenticate', val);
+      },
+    },
+    pickingFromScrap() {
+      return this.$store.state.game.pickingFromScrap;
+    },
+    playerTwoCount() {
+      return this.twoCount(this.player);
+    },
+    playerWins() {
+      return (
+        this.$store.state.game.gameIsOver &&
+        this.$store.state.game.winnerPNum === this.$store.state.game.myPNum
+      );
+    },
+    scrap() {
+      return this.game.scrap;
+    },
+    secondCard() {
+      return this.$store.state.game.secondCard;
+    },
+    showCannotCounterDialog() {
+      return (
+        (this.myTurnToCounter && !this.hasTwoInHand) ||
+        (this.myTurnToCounter && this.hasTwoInHand && this.opponentQueenCount > 0)
+      );
+    },
+    showCounterDialog() {
+      return this.myTurnToCounter && this.hasTwoInHand && this.opponentQueenCount === 0;
+    },
+    showSevenDoubleJacksDialog() {
+      return (
+        this.resolvingSeven &&
+        this.topCard.rank === 11 &&
+        this.secondCard.rank === 11 &&
+        (this.opponentPointTotal === 0 || this.opponentQueenCount > 0)
+      );
+    },
+    stalemate() {
+      return this.$store.state.game.gameIsOver && this.$store.state.game.winnerPNum === null;
+    },
+    topCard() {
+      return this.$store.state.game.topCard;
+    },
+    twosInHand() {
+      return this.player.hand.filter((card) => card.rank === 2);
+    },
+    twosPlayed() {
+      return this.game.twos;
+    },
+  },
+  methods: {
```

Comment:
I definitely agree when it comes to the `clearSelection` and `handleError` methods, I've converted those to emitters in the latest pr. As for the others that i moved down I think they should stay here until they're moved to the store as actions or something similar to keep the `GameView` a bit more slim because it is _C H O N K Y_. Does that work?